source/common/router/config_impl.cc

```diff
@@ -337,8 +438,9 @@ const Route* VirtualHostImpl::getRouteFromEntries(const Http::HeaderMap& headers
 
   // Check for a route that matches the request.
   for (const RouteEntryImplBasePtr& route : routes_) {
-    if (route->matches(headers, random_value)) {
-      return route.get();
+    const Route* cEntry = route->matches(headers, random_value);
```

Comment:
That would throw an error since route is the iteration variable in the for loop. I could rename it to something more sensible.